processing/src/main/java/org/apache/druid/query/scan/ScanResultValue.java

```diff
@@ -78,9 +79,16 @@ public Object getEvents()
   public long getFirstEventTimestamp(ScanQuery.ResultFormat resultFormat)
   {
     if (resultFormat.equals(ScanQuery.ResultFormat.RESULT_FORMAT_LIST)) {
-      return (Long) ((Map<String, Object>) ((List<Object>) this.getEvents()).get(0)).get(ColumnHolder.TIME_COLUMN_NAME);
+      Long timestamp = (Long) ((Map<String, Object>) ((List<Object>) this.getEvents()).get(0)).get(ColumnHolder.TIME_COLUMN_NAME);
+      if (timestamp == null) {
+        throw new ISE("Unable to compare timestamp for rows without a time column");
```

Comment:
This restriction should be more prominent:

- Check it in ScanQuery's constructor, so the Broker will reject the query if it's not going to work.
- Add a note to the documentation that you must include the `__time` column if you are doing time ordering.